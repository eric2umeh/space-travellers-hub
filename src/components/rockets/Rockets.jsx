import React from 'react';
import { useSelector } from 'react-redux';
import RocketsItem from '../rocketsItem/RocketsItem';
import classes from './Rockets.module.css';

const Rockets = () => {
  const rockets = useSelector((state) => state.rockets.rockets);

  return (
    <section className={classes.rocket__list}>
      {rockets.map((item) => (
        <RocketsItem item={item} key={item.rocket_id} />
      ))}
    </section>
  );
};

export default Rockets;
