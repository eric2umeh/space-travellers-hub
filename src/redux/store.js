import { configureStore } from '@reduxjs/toolkit';
import rockets from './rockets/rocketsSlice';

const store = configureStore({
  reducer: {
    rockets: rockets.reducer,
  },
});

export default store;
