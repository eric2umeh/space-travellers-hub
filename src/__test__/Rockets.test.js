import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Rockets from '../components/rockets/Rockets';

describe('testing if rockets data is rendering', () => {
  test('component test', () => {
    const test = renderer.create(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );
    expect(test).toMatchSnapshot();
  });
});
