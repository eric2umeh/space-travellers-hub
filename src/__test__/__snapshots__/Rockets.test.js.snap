// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing if rockets data is rendering component test 1`] = `
<section
  className="rocket__list"
/>
`;
